### PR TITLE
provide test collections as Julia artifacts

### PR DESCRIPTION
**Pull request by dpo**
artifacts are only available in Julia ≥ 1.3

### CODE REVIEW

**Comment by mtanneau** [comment]
Test collections are not needed for the reader itself, are they? Just for benchmark? Wouldn't this fit better in `/benchmarks/`?

I'm not that familiar with the inner working of Artifacts. I see each collection is downloaded as one single compressed file. Are instance files extracted automatically, or does the user have to do it?

**Comment by dpo** [comment]
They're not needed for the reader itself, but those are typical collections that one is likely to use, so it makes sense to me to ship them with the reader. The advantage of artifacts is that they can be shared by packages. If another package uses the Artifacts.toml file, the collections should not (in theory) be downloaded twice.

All instances are extracted. There's an example in the tests. The collections are under `netlib_path` and `mm_path` and you can just `readqps(joinpath(netlib_path, "AFIRO.SIF"))`.

**Comment by mtanneau** [comment]
Looks good to me.

File: src/QPSReader.jl [hunk 1]
```diff
@@ -4,4 +4,29 @@ export QPSData, readqps
...
+
+  ensure_artifact_installed("marosmeszaros", artifact_toml)
+  mm_hash = artifact_hash("marosmeszaros", artifact_toml)
+  @assert artifact_exists(mm_hash)
+
+  @doc "Path to the Maros-Meszaros quadratic optimization test set."
+  const mm_path = joinpath(artifact_path(mm_hash), "optrove-maros-meszaros-9adfb5707b1e")
+
+  # const netlib_path = joinpath(artifact"netliblp", "optrove-netlib-lp-f83996fca937")
+  # const mm_path = joinpath(artifact"marosmeszaros", "optrove-maros-meszaros-9adfb5707b1e")
```
Comment (mtanneau) [inline src/QPSReader.jl:29]: Maybe just remove those two lines?

Comment (dpo) [reply src/QPSReader.jl:29]: In principle, only those two lines are necessary (they are supposed to do the same as the stuff above). But for some reason, they return an error. I asked on Slack to see if anyone has an idea.

**Comment by dpo** [comment]
Any other collection to add? Those artifacts are "lazy", which means they're not downloaded until the user tries to access them.

**Comment by mtanneau** [comment]
> Any other collection to add? Those artifacts are "lazy", which means they're not downloaded until the user tries to access them.

I think these two are enough for now.
I will update the benchmark scripts once the PR is merged.

**Pull request merged**